pkg/ottl/parser.go

```diff
@@ -195,6 +196,33 @@ func (p *Parser[K]) ParseCondition(condition string) (*Condition[K], error) {
 	}, nil
 }
 
+// appendStatementPathsContext changes the given OTTL statement adding the context name prefix
+// to all context-less paths. No modifications are performed for paths which [Path.Context]
+// value matches any WithPathContextNames value.
+// The context argument must be valid WithPathContextNames value, otherwise an error is returned.
+func (p *Parser[K]) appendStatementPathsContext(context string, statement string) (string, error) {
```

Comment:
```suggestion
func (p *Parser[K]) prependContextToStatementPaths(context string, statement string) (string, error) {
```
Not required for this PR, and a little minor, but looking at this again, I think this is a bit more descriptive name.

Reply:
Agreed, I'll rename it, thanks!